src/service/ConfigService.ts

```diff
@@ -692,6 +692,21 @@ export class ConfigService {
                     [],
                     ['node.crt.pem', 'node.key.pem', 'ca.cert.pem'],
                 );
+
+                if (gatewayPreset.restProtocol === 'HTTPS') {
+                    if (gatewayPreset.restSSLKeyBase64 && gatewayPreset.restSSLCertificateBase64) {
```

Comment:
could you try what happens to the file when you --upgrade with and without providing the custom preset?